### DISCUSSION TITLE
電源の設定

### DISCUSSION THREAD

**Discussion by bbm-soka** [ponoor/step-series-support#34] [General]
step400デバイスでNEMA24サイズのステッピングモーター(https://www.omc-stepperonline.com/ja/pシリーズnema24ステッピングモーター3nm-424-92oz-in-電磁ブレーキ付き-24bk20-30)を電圧モードにて4台制御しようとしています。
負荷を吊り下げて使用する想定です。

この場合24V10Aのスイッチング電源で安定して動作させることは可能でしょうか？

何か電源の選定をする上で、気にしておく項目などあればご教示いただけると嬉しいです。

**Comment by kanta** [comment]
相電流の定格が大きなモータですね。
必要なスピードやトルク次第で大きく変わってくるのでやってみないと分からない部分は大きいですが、必要なトルク（≒相電流）を流すには電源電圧が足りない可能性はありそうです。特にスピードを上げると、必要な電圧は高くなります。まず24Vで動かしてみて、トルクが不足する場合はスイッチング電源を直列にして48V, 72Vなどに電圧を上げてみるのはどうでしょうか。
電源の電流定格についても上記の回転スピード、トルクによって変わってきますので、実際に電源から流れている電流を実測しながら検討するのが良いかと思います。
またSTEP400自体もかなり発熱すると思いますので、放熱器や冷却ファンの取り付けをご検討ください。

**Reply by bbm-soka** [reply]
ご教示いただきありがとうございます！

今回スピードはあまり出さないのですが、保持トルクをそれなりに上げる必要があるので24Vでは難しそうでした。

24Vで試してみたところ、モーター2つぐらいであれば問題はなさそうでしたが、48Vに変更しモーター4台動かすと"OverCurrent"になってしまいうまく動作せずでした。
対策としては、抵抗などを挟んで電流を下げるなどでしょうか？

**Reply by kanta** [reply]
いま制御は電圧モードと電流モードどちらで使用されていますか？電圧モードで動かしている場合はKVAL, 電流モードで動かしている場合はTVALの値が大きすぎるかもしれません。

**Reply by bbm-soka** [reply]
現在は電圧モードで使用しています。
負荷の重さの関係で、保持トルクをある程度上げる必要があり、下記のような値になっています。

"KVAL_HOLD": 50
"KVAL_RUN": 90
"KVAL_ACC": 90
"KVAL_DEC": 90

もう少しKVALの設定を見直してみます。

48V10Aの電源を繋いだ場合4つの端子にどのくらいの電流が流れるのでしょうか？

**Reply by kanta** [reply]
実際にモータに流れる電流は数多くの要因があって、特に電圧モードの場合は回転数による変化が大きくなかなか予測が難しいように思います。

over currentになる場合は、BEMFパラメータの設定で回避できるかもしれません。基本的にモータの回転が速くなると電流が流せなくなってくるので、目的のスピードでちょうどよいKVALに設定した場合、そのスピードに達するまでの間に電流が流れすぎる可能性があります。
それを補正するために、 `ST_SLP` `FN_SLP_ACC` `FN_SLP_DEC` の値を `255` までの範囲内で大きくしていって、その分 KVALを下げていくと、低速域での過電流が防止できる可能性があります。

https://ponoor.com/docs/step-series/osc-command-reference/voltage-and-current-mode-settings/

**Reply by bbm-soka** [reply]
ありがとうございます。
BEMFパラメータは初期設定から変更しておらずでした。
試してみます。

すいません、あまりBEMFパラメータの動作を理解できておらずで、、
BEMFパラメータの動作としては値を大きくしていくことで、電流の量を調整する効果が上がるというようなイメージでしょうか？

**Reply by kanta** [reply]
BEMF補償のパラメータは、回転数に比例してKVALに効いてくる係数で、値を大きくするとこの傾きが大きくなり、相対的に回転数が大きいときにモータに印加する電圧が高くなります。これはある一定電圧をモータにかけた場合、低速域で電流が流れやすく、高速になるにつれ流れにくくなることから、その差を補うためのものです。
![image](https://github.com/ponoor/step-series-support/assets/1097902/6342cf4a-5d6b-441a-b3e0-ba2453d8685c)

**Reply by bbm-soka** [reply]
ありがとうございます！
なるほどです！なんとなくイメージがつきました。
もう一度設定を見直してみます。
